### DISCUSSION TITLE
jbovlasteのXMLエクスポート取得の自動化について

### DISCUSSION THREAD

**Discussion by niftg** [ponjbogri/jvs_ja#4] [Show and tell]
先日漸く知ったのですが、jbovlasteログイン時のcookieだけあれば、curl等で[`xml-export`ファイル群](https://jbovlaste.lojban.org/export/xml.html)の機械的な取得も可能な様です。

実際、その方法で[バージョン化されたJVS辞書の形成を試みたリポジトリ][jd]が存在しますし、自前でも更に簡易的なスクリプト群を作成してみました：[la ninpikta jo'u la vlastecpe](https://gist.github.com/niftg/f93327c98d090ecea61cf1ce9c6351e2)

もちろんcookieの管理には細心の注意が必要ですが、上手く活用できたらと存じます。

[jd]: https://github.com/lojbanistan/jbovlaste-dicts

**Comment by qq542vev** [comment]
gitやjbovlasteの仕組みについてあまり詳しくないのですが、例えばにふさん制作のスクリプトを何処かのサーバにおいて、スクリプトを定期実行によってXMLを取得して、変更があればその都度コミットするようにすれば、一応は単語の変更箇所の差分が取れたりするのでしょうか。

**Reply by niftg** [reply]
.ie。ただし、diffによる簡易的な差分取得においては、jbovlaste側でXMLの出力順序や方法がころころ変化しないことが前提です。（順序については一応、ある程度吸収できる仕組みも作ってはみたのですが、後ほど）
まあ折角のXMLなので、差分検出時に全走査して変更内容の分析もしてみた方が良いのかもしれません。

**Reply by qq542vev** [reply]
あまり詳しくないのですが、jbovlasteのXMLデータは順序や方法がそんなにころころ変化するしたりするものなのでしょうか。🤔

**Reply by niftg** [reply]
滅多には無いとは存じますが、一年越しのコミット (_https://github.com/ponjbogri/jvs_ja/commit/9751d21ba4baf80c9b3eee71595cb081bdb7158a_) の際に一度経験しました

**Comment by qq542vev** [comment]
<https://github.com/qq542vev/jvs_ja/commits/master>

実験的ではありますが、1日1回英語版と日本語版の xml-export を取得し更新があればコミットするのを行ってみました。

**Reply by niftg** [reply]
これは助かりますね！もし支障無ければ、そちらの自動コミットをこちらで適宜cherry-pickしてmasterブランチに適用しても良いですか？
具体的にはそちらでの[XMLエクスポートの2022-02-12までの更新](https://github.com/qq542vev/jvs_ja/commits/461e394502638dcfedba6296828ef7c010e0960d/xml-export.html.xml)をこちらで再現する形になります。
mu'o .io la'o zoi @qq542vev zoi

**Reply by qq542vev** [reply]
ありがとうございます。当方相変わらず、gitの操作方法について明るくないので、申し訳ないのですが、にふさんの思いのままに、ご自由に利用して頂ければ幸いです。
